buttercup/cogs/rules.py

```diff
@@ -42,41 +114,45 @@ def __init__(self, bot: ButtercupBot, reddit_api: asyncpraw.Reddit) -> None:
     )
     async def _rules(self, ctx: SlashContext, subreddit: str) -> None:
         """Get the rules of the specified subreddit."""
-        # Send a quick response
-        # We will edit this later with the actual content
         start = datetime.now()
         sub_name = extract_sub_name(subreddit)
+        # Send a quick response
+        # We will edit this later with the actual content
         msg = await ctx.send(i18n["rules"]["getting_rules"].format(sub_name))
 
-        sub = await self.reddit_api.subreddit(sub_name)
+        rules = await self._get_rule_list(msg, subreddit)
+        if rules is None:
+            return
+
+        await send_rules_message(msg, rules, subreddit, start, "rules")
 
-        embed = Embed(title=i18n["rules"]["embed_title"].format(sub_name))
+    @cog_ext.cog_slash(
+        name="pirules",
+        description="Get the rules of the specified subreddit "
+        "regarding personal information.",
+        options=[
+            create_option(
+                name="subreddit",
+                description="The subreddit to get the PI rules of.",
+                option_type=3,
+                required=True,
+            )
+        ],
+    )
+    async def _pi_rules(self, ctx: SlashContext, subreddit: str) -> None:

```

Comment:
I think the split really works, but that the return value of None resembling a message being sent could still be clearer in my opinion. What would you think of making a bigger part generic, e.g. something like:

```
async def _filtered_rules(self, ctx: SlashContext, subreddit: str, localization_key: str,
                          filter_function: Callable[[Rule], Boolean]) -> None:
    start = datetime.now()
    sub_name = extract_sub_name(subreddit)
    # Send a quick response
    # We will edit this later with the actual content
    msg = await ctx.send(i18n[localization_key]["getting_rules"].format(sub_name))
    try:
        sub = await self.reddit_api.subreddit(subreddit)
        return send_rules_message(msg, [rule async for rule in sub.rules if filter_function(rule)], start, localization_key)
    except Redirect:
        # The subreddit does not exist
        await msg.edit(content=i18n["rules"]["sub_not_found"].format(subreddit))
    except NotFound:
        # A character in the sub name is not allowed
        await msg.edit(content=i18n["rules"]["sub_not_found"].format(subreddit))
    except Forbidden:
        # The subreddit is private
        await msg.edit(content=i18n["rules"]["sub_private"].format(subreddit))

async def _rules(self, ctx: SlashContext, subreddit: str) -> None:
    _filtered_rules(ctx, subreddit, "rules", lambda rule: True)

async def _pi_rules(self, ctx: SlashContext, subreddit: str) -> None:
    _filtered_rules(ctx, subreddit, "rules", is_pi_rule)
```

Where the bottom two functions would be the slash commands. This would also result in the `get_result_list` method being unused.

Reply:
Good idea. Fixed!